### DISCUSSION TITLE
Multiple Leader Keys

### DISCUSSION THREAD

**Discussion by AH-Merii** [neovim/neovim#30046] [Q&A] [answered]
Is it possible to setup multiple leader keys?

What I would like to do is make `s` my other "leader" key, so that when I use the plugin [`mini.surroud`](https://github.com/echasnovski/mini.surround) I am able to access the other mappings as well. 

Currently, I need to click `s` followed by ,for example, `a` really quickly in order to activate `mini.surround` otherwise it will append. I know one workaround is to increase the `timoutlen`, but that feels like a hack. Other solutions are also welcome.

**Comment by zeertzjq** [comment]
What does this have to do with leader keys? A leader key is just a global variable.

**Reply by AH-Merii** [reply]
I understand that, but keymaps using the leader key as prefix behave differently than keymaps using a normal character as prefix.

When I enter <leader>, I am able to wait as long as I want before entering the any other key, however, when I use a normal character, I have to enter the next key immediately, otherwise, it will timeout, and revert back to the normal behaviour of the second key.

For more context, I am using `whichkey`, do you think that may be altering the way the leader key behaves?

**Reply by zeertzjq** [reply, marked as answer]
It seems that which-key.nvim doesn't allow `s` as a trigger. I think you should open a feature request at which-key.nvim repository to allow unsafe triggers. Or you can just modify the source code of which-key.nvim to allow that:
```diff
diff --git a/lua/which-key/buf.lua b/lua/which-key/buf.lua
index 9f4a501..2c122ce 100644
--- a/lua/which-key/buf.lua
+++ b/lua/which-key/buf.lua
@@ -26,7 +26,7 @@ local function is_safe(node, no_single)
   if no_single and #node.path == 1 then
     local key = node.path[1]
     -- only z or g are safe
-    if key:match("^[a-z]$") and not key:match("^[gz]$") then
+    if key:match("^[a-z]$") and not key:match("^[gsz]$") then
       return false
     end
     -- only Z is safe
```

**Reply by AH-Merii** [reply]
> It seems that which-key.nvim doesn't allow `s` as a trigger. I think you should open a feature request at which-key.nvim repository to allow unsafe triggers. Or you can just modify the source code of which-key.nvim to allow that:
> ```diff
> diff --git a/lua/which-key/buf.lua b/lua/which-key/buf.lua
> index 9f4a501..2c122ce 100644
> --- a/lua/which-key/buf.lua
> +++ b/lua/which-key/buf.lua
> @@ -26,7 +26,7 @@ local function is_safe(node, no_single)
>    if no_single and #node.path == 1 then
>      local key = node.path[1]
>      -- only z or g are safe
> -    if key:match("^[a-z]$") and not key:match("^[gz]$") then
> +    if key:match("^[a-z]$") and not key:match("^[gsz]$") then
>        return false
>      end
>      -- only Z is safe
> ```

Thank you so much for this, I really appreciate you taking the time to help me out!
